pkg/agent/controller/egress/egress_controller.go

```diff
@@ -745,6 +768,9 @@ func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP s
 					Message:            fmt.Sprintf("Failed to assign the IP to EgressNode: %v", scheduleErr),
 				},
 			}
+			if c.record != nil {
+				c.record.Eventf(egress, corev1.EventTypeNormal, "IPUnassigned", "Unassigned egress %s with IP %s", egress.Name, egress.Spec.EgressIP)
```

Comment:
I believe this will cause multiple events which I suggested to avoid exactly.
It should only send the event when it's the chosen agent. See how `updateEgressStatus` determines which one should update the Egress's status.